src/autosuggest/dropdown-controller.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useState } from 'react';
+
+export interface UseAutosuggestDropdownProps {
+  readOnly?: boolean;
+  onClose?: () => void;
+}
+
+export interface AutosuggestDropdownState {
+  open: boolean;
+}
+
+export interface AutosuggestDropdownHandlers {
+  openDropdown(): void;
+  closeDropdown(): void;
+}
+
+export const useAutosuggestDropdown = ({
```

Comment:
minor: Not sure what's the Autosuggest-specific part. It looks more like a generic dropdown state management. `readOnly` could also be differently named.

Reply:
I would prefer keeping it here to allow for more extensions that are specific to autosuggest, e.g. the conditional focus-lock and more.

Reply:
Added more logic in there for the intent to be better clear.